src/main/java/transact/model/transaction/info/TransactionId.java

```diff
@@ -3,55 +3,81 @@
 import static java.util.Objects.requireNonNull;
 import static transact.commons.util.AppUtil.checkArgument;
 
-import transact.commons.util.StringUtil;
+import java.util.TreeSet;
 
 /**
  * Represents a unique transaction ID in the system.
- * Guarantees: immutable; is valid as declared in
- * {@link #isValidTransactionId(String)}
+ * Guarantees: immutable; is valid as declared in constructors
  */
 public class TransactionId {
 
-    public static final String MESSAGE_CONSTRAINTS = "Transaction ID should be an 8-character alphanumeric string";
-    public static final String VALIDATION_REGEX = "[A-Za-z0-9]{8}";
-    public final String value;
-    public final int transactionIdLength = 8;
+    public static final String MESSAGE_CONSTRAINTS = "Transaction ID should be a unique number";
+    public static final String VALIDATION_REGEX = "^\\d+$";
+
+    private static TreeSet<Integer> usedIds = new TreeSet<>();
+
+    public final Integer value;
 
     /**
      * Constructs a {@code TransactionId}.
      *
      */
     public TransactionId() {
-        value = StringUtil.generateRandomString(transactionIdLength);
-        requireNonNull(value);
-        checkArgument(isValidTransactionId(value), MESSAGE_CONSTRAINTS);
+        this(usedIds.isEmpty() ? 0 : usedIds.last() + 1);
     }
 
     /**
      * Constructs a {@code TransactionId} with specified {@code id}
      */
-    public TransactionId(String id) {
+    public TransactionId(Integer id) {
         requireNonNull(id);
-        checkArgument(isValidTransactionId(id), MESSAGE_CONSTRAINTS);
+        checkArgument(!usedIds.contains(id), MESSAGE_CONSTRAINTS);
```

Comment:
```suggestion
        checkArgument(isValidTransactionId(id), MESSAGE_CONSTRAINTS);
```
I think don't need change since we also updated the method